examples/server/server.cpp

```diff
@@ -384,7 +384,8 @@ struct server_task {
                             SRV_DBG("Grammar trigger token: %d (`%s`)\n", token, word.c_str());
                             common_grammar_trigger trigger;
                             trigger.type = COMMON_GRAMMAR_TRIGGER_TYPE_TOKEN;
-                            trigger.value = (llama_token) token;
+                            trigger.value = word;
+                            trigger.token = token;
                             params.sampling.grammar_triggers.push_back(trigger);
```

Comment:
Save a string copy here:

```suggestion
                            params.sampling.grammar_triggers.push_back(std::move(trigger));
```